From c738d2eab8606e69cd9d4aabf2150b01f3f4f10b Mon Sep 17 00:00:00 2001
Message-Id: <c738d2eab8606e69cd9d4aabf2150b01f3f4f10b.1421272770.git.jen@redhat.com>
In-Reply-To: <acd6d327ae856e72f084949fd71314237a4b9b8b.1421272770.git.jen@redhat.com>
References: <acd6d327ae856e72f084949fd71314237a4b9b8b.1421272770.git.jen@redhat.com>
From: Juan Quintela <quintela@redhat.com>
Date: Wed, 7 Jan 2015 16:45:10 -0600
Subject: [CHANGE 06/10] migration: remove not needed ram_save_remaining
 function
To: rhvirt-patches@redhat.com,
    jen@redhat.com

RH-Author: Juan Quintela <quintela@redhat.com>
Message-id: <1420649114-17435-7-git-send-email-quintela@redhat.com>
Patchwork-id: 63163
O-Subject: [PATCH qemu-kvm RHEL6.7 06/10] migration: remove not needed ram_save_remaining function
Bugzilla: 970103
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Dr. David Alan Gilbert (git) <dgilbert@redhat.com>

We were really needing ram_bytes_remaining() anyways.

Signed-off-by: Juan Quintela <quintela@redhat.com>
---
 vl.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

Signed-off-by: Jeff E. Nelson <jen@redhat.com>
---
 vl.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/vl.c b/vl.c
index 01a480a..e5d239a 100644
--- a/vl.c
+++ b/vl.c
@@ -2833,14 +2833,9 @@ static int ram_save_block(QEMUFile *f)
 
 static uint64_t bytes_transferred;
 
-static ram_addr_t ram_save_remaining(void)
-{
-    return ram_list.dirty_pages;
-}
-
 uint64_t ram_bytes_remaining(void)
 {
-    return ram_save_remaining() * TARGET_PAGE_SIZE;
+    return ram_list.dirty_pages * TARGET_PAGE_SIZE;
 }
 
 uint64_t ram_bytes_transferred(void)
@@ -2959,7 +2954,7 @@ static int ram_save_live(Monitor *mon, QEMUFile *f, int stage, void *opaque)
     if (stage == 2) {
         uint64_t expected_time;
 
-        expected_time = ram_save_remaining() * TARGET_PAGE_SIZE / bwidth;
+        expected_time = ram_bytes_remaining() / bwidth;
         return expected_time <= migrate_max_downtime();
     }
     return 0;
-- 
2.1.0

